cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -1075,9 +1098,61 @@ std::vector<AtomKey> LocalVersionedEngine::batch_write_internal(
             validate_index
         ));
     }
-    return folly::collect(results_fut).get();
+    return results_fut;
+}
+
+std::pair<VersionId, std::shared_ptr<DeDupMap>> LocalVersionedEngine::create_version_id_and_dedup_map(
```

Comment:
Prefer a struct to `std::pair`, it makes intent much clearer. It will also help with passing generated values between tasks, eliminating the need for `vector_update_info`.

Reply:
yes, it will help, but only for the first passing level of tasks (regarding your comment about vector_update_info)